### PR TITLE
docs: add xrpl.js code snippets over to xrpl-py

### PR DESCRIPTION
**Pull request by connorjchen**
## High Level Overview of Change

Ported xrpl.js code snippets over to xrpl-py

### Context of Change

Code snippets are a great way for developers in the community to learn from

### Type of Change

- [x] Documentation Updates

## Test Plan

Ran each file individually **(claim_payment_channel fails because missing hashing function now)**

### CODE REVIEW

**Review by mvadari — COMMENTED** [review]
I would write these with the `JsonRpcClient` (unless you want to write a snippet with each client, which seems like too much).

File: snippets/get_transaction.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    """
+    Async snippet that walks us through getting a transaction.
+
+    Args:
+        client: The async network client to use to send the request.
+
+    Raises:
+        Exception: if meta not included in the transaction response.
+    """
+    await client.open()
```
Comment (mvadari) [inline snippets/get_transaction.py:16]: If you do want to use the websocket clients, you should use `with ... as ...` syntax - the websocket client documentation explains details.

File: snippets/get_transaction.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    Async snippet that walks us through getting a transaction.
+
+    Args:
+        client: The async network client to use to send the request.
+
+    Raises:
+        Exception: if meta not included in the transaction response.
+    """
+    async with AsyncWebsocketClient("wss://s.altnet.rippletest.net:51233") as client:
+        await client.open()
```
Comment (mvadari) [inline snippets/get_transaction.py:17]: You don't need to open and close the client when you use `with ... as ...` syntax - that does it automatically for you.

File: snippets/get_transaction.py [hunk 3]
```diff
@@ -1,49 +1,47 @@
 """A snippet that walks us through getting a transaction."""
-from xrpl.asyncio.clients import AsyncWebsocketClient
+from xrpl.clients import JsonRpcClient
 from xrpl.models.requests import Ledger, Tx
 
 
-async def async_get_transaction() -> None:
+def get_transaction(client: JsonRpcClient) -> None:
```
Comment (mvadari) [inline snippets/get_transaction.py:6]: Why did you move the client to outside the function?

Comment (mvadari) [reply snippets/get_transaction.py:6]: Actually, I don't know if these have to be functions at all - I think these could just be scripts.

Comment (connorjchen) [reply snippets/get_transaction.py:6]: If I don't make it a function, I think it tries running it when I make docs

Comment (connorjchen) [reply snippets/get_transaction.py:6]: Also, I can move client into function. No real reason I put it outside.

Comment (connorjchen) [reply snippets/get_transaction.py]: Actually, I think I can make it not run on make docs

File: docs/source/snippets.rst [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+.. toctree::
+   :maxdepth: 1
+
+   get_transaction
+   partial_payment
+   paths
+   reliable_transaction_submission
+   send_escrow
+   set_regular_key
```
Comment (JST5000) [inline docs/source/snippets.rst:14]: Order matters here :) - We want simpler cases to the top in my opinion
```suggestion
   get_transaction
   reliable_transaction_submission
   set_regular_key
   send_escrow
   paths
   partial_payment
```

Comment (JST5000) [inline docs/source/snippets.rst:14]: Also, is it possible to give these more human readable names?

Ex. 
* Look up a transaction on ledger
* Send a transaction and see it get validated
* Change your password by setting a regular key
* Set up an Escrow
* Find the best path to trade with
* Handle partial payments

File: snippets/get_transaction.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
+"""A snippet that walks us through getting a transaction."""
```
Comment (JST5000) [inline snippets/get_transaction.py:1]:
```suggestion
"""How to see a validated transaction"""
```
(Or something else more active + about the lesson rather than the implementation)

File: snippets/get_transaction.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+from xrpl.clients import JsonRpcClient
+from xrpl.models.requests import Ledger, Tx
+
+client = JsonRpcClient("https://s.altnet.rippletest.net:51234/")
```
Comment (JST5000) [inline snippets/get_transaction.py:5]: Can you add comments as we go explaining why we're doing things? 

Ex here: # First we connect to the test network

File: snippets/get_transaction.py [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+ledger_request = Ledger(transactions=True, ledger_index="validated")
+ledger_response = client.request(ledger_request)
+print(ledger_response)
+
+transactions = ledger_response.result["ledger"]["transactions"]
+
+if transactions:
```
Comment (JST5000) [inline snippets/get_transaction.py:13]: What happens if there are no transactions? (Ex. Will often close a ledger with no transactions)
https://devnet.xrpl.org/

File: snippets/get_transaction.py [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+    # delivered_amount is the amount actually received by the destination account.
+    # Use this field to determine how much was delivered,
+    # regardless of whether the transaction is a partial payment.
+    # https://xrpl.org/transaction-metadata.html#delivered_amount
+    if type(tx_response.result["meta"] != "string"):
+        if "delivered_amount" in tx_response.result["meta"]:
+            print("delivered_amount:", tx_response.result["meta"]["delivered_amount"])
+        else:
+            print("delivered_amount: undefined")
```
Comment (JST5000) [inline snippets/get_transaction.py:31]: This all seems like a lot of extra complexity that's not required for a `get_transaction` snippet. I'd probably just remove this.

File: snippets/partial_payment.py [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+from xrpl.models.transactions import Payment, PaymentFlag, TrustSet
+from xrpl.transaction import (
+    safe_sign_and_autofill_transaction,
+    send_reliable_submission,
+)
+from xrpl.wallet import generate_faucet_wallet
+
+client = JsonRpcClient("https://s.altnet.rippletest.net:51234/")
+
+# creating wallets as prerequisite
```
Comment (JST5000) [inline snippets/partial_payment.py:14]:
```suggestion
# Creating two wallets to send money between
```

Style nit: Comments should start with a capital letter (Ex. creating -> Creating)

File: docs/source/snippets.rst [hunk 10]
```diff
@@ -0,0 +1,14 @@
+XRPL Code Snippets
+===================
+
+Code snippets to demonstrate basic usage of the xrpl-py library.
+
+.. toctree::
+   :maxdepth: 1
+
+   get_transaction
```
Comment (mvadari) [inline docs/source/snippets.rst:9]: Should there be links to each snippet, or should all the snippets just be on this page?

File: snippets/get_transaction.py [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+# References
+# - https://xrpl.org/look-up-transaction-results.html
+# - https://xrpl.org/parallel-networks.html#parallel-networks
+# - https://xrpl.org/tx.html
+
+# Create a client to connect to the test network
+client = JsonRpcClient("https://s.altnet.rippletest.net:51234/")
+
+# Create a Ledger request and have the client call it
+ledger_request = Ledger(transactions=True, ledger_index="validated")
```
Comment (mvadari) [inline snippets/get_transaction.py:14]: super nit: IMO the params in this object should be flipped in order

File: snippets/get_transaction.py (hunk 7 above)
Comment (mvadari) [reply snippets/get_transaction.py:23]: This specific snippet should probably be on mainnet, then - otherwise you'll mostly get nothing

**Review by mvadari — COMMENTED** [review]
A couple of notes:
* Please add a multisign snippet (this could be in a separate PR if you'd prefer)
* Please add these as tests (this would probably best done by importing the file in a test, since the file will run when it's imported)

File: docs/source/snippets.rst (hunk 10 above)
Comment (connorjchen) [reply docs/source/snippets.rst:9]: Hm yeah I think all snippets on one page might be bit better and still have tabs on left bar for it

File: docs/source/snippets.rst [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+-------------------------------------
+
+.. literalinclude:: ../../snippets/get_transaction.py
+
+Send a Transaction and See if It Gets Validated
+-----------------------------------------------
+
+.. literalinclude:: ../../snippets/reliable_transaction_submission.py
+
+Change Your Password by Setting a Regular Key
```
Comment (mvadari) [inline docs/source/snippets.rst:19]: I don't really like the phrase "password" here - can we leave the title at `Set a Regular Key`?

File: snippets/set_regular_key.py [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+)
+from xrpl.wallet import generate_faucet_wallet
+
+# References
+# - https://xrpl.org/assign-a-regular-key-pair.html#assign-a-regular-key-pair
+# - https://xrpl.org/setregularkey.html#setregularkey
+# - https://xrpl.org/change-or-remove-a-regular-key-pair.html#change-or-remove-a-regular-key-pair
+
+# Create a client to connect to the test network
+client = JsonRpcClient("https://s.altnet.rippletest.net:51234/")
```
Comment (mvadari) [inline snippets/set_regular_key.py:17]: nit: leave off the `/` at the end

**Review by mvadari — APPROVED** [review]

**Review by jonathanlei — APPROVED** [review]
